testsuite/tests/typing-modules/records_errors_test.ml

```diff
@@ -0,0 +1,138 @@
+(* TEST
+ * expect
+*)
+
+module M1 : sig
+  type t = {f0 : unit * unit * unit * int * unit * unit * unit;
+            f1 : unit * unit * unit * int * unit * unit * unit}
+end = struct
+  type t = {f0 : unit * unit * unit * float* unit * unit * unit;
+            f1 : unit * unit * unit * string * unit * unit * unit}
+end;;
+[%%expect{|
+Lines 4-7, characters 6-3:
+4 | ......struct
+5 |   type t = {f0 : unit * unit * unit * float* unit * unit * unit;
+6 |             f1 : unit * unit * unit * string * unit * unit * unit}
+7 | end..
+Error: Signature mismatch:
+       Modules do not match:
+         sig
+           type t = {
+             f0 : unit * unit * unit * float * unit * unit * unit;
+             f1 : unit * unit * unit * string * unit * unit * unit;
+           }
+         end
+       is not included in
+         sig
+           type t = {
+             f0 : unit * unit * unit * int * unit * unit * unit;
+             f1 : unit * unit * unit * int * unit * unit * unit;
+           }
+         end
+       Type declarations do not match:
+         type t = {
+           f0 : unit * unit * unit * float * unit * unit * unit;
+           f1 : unit * unit * unit * string * unit * unit * unit;
+         }
+       is not included in
+         type t = {
+           f0 : unit * unit * unit * int * unit * unit * unit;
+           f1 : unit * unit * unit * int * unit * unit * unit;
+         }
+       Fields do not match:
+         f0 : unit * unit * unit * float * unit * unit * unit;
+       is not compatible with:
+         f0 : unit * unit * unit * int * unit * unit * unit;
+       The types are not equal.
```

Comment:
I am a bit surprised by this example. It is quite lengthy, however, this lenght does not highlight the best point of this PR, since there are only two fields (with a long title) rather than a long list of fields. Is there an incoming PR that refines the error in this case?

Reply:
Yes, the next step is to replace equal, moregenal and subtype functions in Ctype by new, which throw exception with error trace instead of returning false (like unify). With error trace it is possible to zoom the error and tell which part exactly is causing mismatch. For the case above additional output for now is:
```
Type unit * unit * unit * float * unit * unit * unit
is not compatible with type
    unit * unit * unit * int * unit * unit * unit
Type float is not compatible with type int
```

You can get more details [here](https://github.com/MekhrubonT/ocaml/pull/2/files)